packages/zoe/src/typeGuards.js

```diff
@@ -79,3 +121,251 @@ export const InvitationElementShape = M.splitRecord({
+    ProposalShape,
+    ExitObjectShape,
+    SeatShape,
+  ).returns(SeatShape),
+  exitAllSeats: M.call(M.any()).returns(),
+  failAllSeats: M.call(M.any()).returns(),
+  exitSeat: M.call(SeatShape, M.any()).returns(),
+  failSeat: M.call(SeatShape, M.any()).returns(),
+  makeZoeMint: M.call(KeywordShape)
+    .optional(
+      AssetKindShape,
+      DisplayInfoShape,
+      M.splitRecord(harden({}), harden({ elementShape: M.pattern() })),
+    )
+    .returns(M.remotable('zoeMint')),
+  registerFeeMint: M.call(KeywordShape, M.remotable('feeMintAccess')).returns(
+    M.remotable('feeMint'),
+  ),
+  replaceAllocations: M.call(SeatHandleAllocationsShape).returns(),
+  stopAcceptingOffers: M.call().returns(),
+  setOfferFilter: M.call(M.arrayOf(M.string())).returns(),
+  getOfferFilter: M.call().returns(M.arrayOf(M.string())),
+  getExitSubscriber: M.call(SeatShape).returns(SubscriberShape),
+  isBlocked: M.call(M.string()).returns(M.boolean()),
+});
+
+export const InstanceStorageManagerIKit = harden({
+  instanceStorageManager: M.interface('InstanceStorageManager', {
+    getTerms: M.call().returns(M.splitRecord(TermsShape)),
+    getIssuers: M.call().returns(IssuerKeywordRecordShape),
+    getBrands: M.call().returns(BrandKeywordRecordShape),
+    getInstallationForInstance: M.call().returns(InstallationShape),
+    getInvitationIssuer: M.call().returns(IssuerShape),
+
+    saveIssuer: M.call(IssuerShape, KeywordShape).returns(M.promise()),
+    makeZoeMint: M.call(KeywordShape)
+      .optional(
+        AssetKindShape,
+        DisplayInfoShape,
+        M.splitRecord(harden({}), harden({ elementShape: M.pattern() })),
+      )
+      .returns(M.eref(ZoeMintShape)),
+    registerFeeMint: M.call(KeywordShape, M.remotable('feeMintAccess')).returns(
+      M.remotable('feeMint'),
+    ),
+    getInstanceRecord: M.call().returns(InstanceRecordShape),
+    getIssuerRecords: M.call().returns(M.arrayOf(IssuerRecordShape)),
+    getWithdrawFacet: M.call().returns(M.remotable('WithdrawFacet')),
+    initInstanceAdmin: M.call(
+      InstanceHandleShape,
+      M.remotable('instanceAdmin'),
+    ).returns(M.promise()),
+    deleteInstanceAdmin: M.call(InstanceAdminI).returns(),
+    makeInvitation: M.call(InvitationHandleShape, M.string())
+      .optional(M.record(), M.pattern())
+      .returns(M.promise()),
+    getRoot: M.call().returns(M.any()),
+    getAdminNode: M.call().returns(M.remotable('adminNode')),
+  }),
+  withdrawFacet: M.interface('WithdrawFacet', {
+    withdrawPayments: M.call(AmountKeywordRecordShape).returns(
+      PaymentPKeywordRecordShape,
+    ),
+  }),
+});
+
+export const BundleCapShape = M.remotable('bundleCap');
+export const SourceBundleShape = M.recordOf(
+  M.string(),
+  M.string({ stringLengthLimit: Infinity }),
+);
+export const ModuleFormatBundleShape = M.splitRecord(
+  harden({ moduleFormat: M.any() }),
+);
+export const BundleShape = M.and(ModuleFormatBundleShape, SourceBundleShape);
```

Comment:
```suggestion
export const BundleShape = M.and(
  M.splitRecord(harden({ moduleFormat: M.string() })),
  M.recordOf(
    M.string(),
    M.string(harden({ stringLengthLimit: Infinity })),
  ),
);
```

How does `splitRecord` differ from `partial`? Is `harden` necessary there in a way that would apply  equally to `M.string`’s arg?

The bundle shape described in this way applies equally well to source bundles and hash bundles, so the intermediate names are not useful. If we wish to cut finer distinctions, we’ll need shapes for `SourceBundle` and `HashBundle` that are the union of supported formats in each case. HashBundle is of the form `{ moduleFormat: 'endoZipBase64Sha512', endoZipBase64Sha512: string }` and SourceBundle is one of `{ moduleFormat: 'getExports', source: string }`, `{ moduleFormat: 'nestedEvaluate', source: string }`, or `{ moduleFormat: 'endoZipBase64', endoZipBase64: string, endoZipBase64Sha512?: string }`.

Reply:
I switched to this simplified `BundleShape`, and used it in all the places `BundleShape` or `SourceBundleShape` appeared. I'll merge like this.